bwameth.py

```diff
@@ -31,10 +32,35 @@
 except ImportError: # python3
     izip = zip
     maketrans = str.maketrans
+import toolshed
 from toolshed import nopen, reader, is_newer_b
 
 __version__ = "0.2.0"
 
+def nopen_keep_parent_stdin(f, mode="r"):
+
+    if f.startswith("|"):
+        # using shell explicitly makes things like process substitution work:
+        # http://stackoverflow.com/questions/7407667/python-subprocess-subshells-and-redirection
+        # use sys.stderr so we dont have to worry about checking it...
+        p = Popen(f[1:], stdout=PIPE, stdin=sys.stdin,
+                  stderr=sys.stderr if mode == "r" else PIPE,
+                  shell=True, bufsize=-1, # use system default for buffering
+                  preexec_fn=toolshed.files.prefunc,
+                  close_fds=False, executable=os.environ.get('SHELL'))
+        if sys.version_info[0] > 2:
+            import io
+            p.stdout = io.TextIOWrapper(p.stdout)
+            p.stdin = io.TextIOWrapper(sys.stdin)
```

Comment:
shouldn't this be:
```
p.stdin = io.TextIOWrapper(p.stdin)
```
?

Reply:
I [originally tried](https://github.com/bwlang/bwa-meth/commit/0062c169f0490ab685c3483b19c6a91f39e276cf) p.stdin, but that returned null - had to switch to sys.stdin to resolve the issue. not sure I understand why p.stdin was null though.